### DISCUSSION TITLE
Which versions support microSD storage

### DISCUSSION THREAD

**Discussion by kevinmcmurtrie** [osmandapp/OsmAnd#15175] [Q&A]
Which versions of OsmAnd+ support microSD storage?  I only know:
- Play Store : No.  Hangs because of [SAF is too slow](https://issuetracker.google.com/issues/179412245)
- F-Droid : Yes, after granting "Allow management of all files" permission and moving the files

I'm using the F-Droid version but features are lagging or disabled.  Notably, the donation features are not connected to anything because it's not part of a commercial store.  Has anyone tried the others?

**Comment by dmpr0** [comment]
The current version supports External storage.

**Reply by kevinmcmurtrie** [reply]
Current version of what?

**Reply by dmpr0** [reply]
OsmAnd

**Reply by kevinmcmurtrie** [reply]
I'm asking about distribution versions.  There's Play Store (no), F-Droid (yes), Amazon (?), Huawei (?), apk (no pro/+)

**Reply by dmpr0** [reply]
I have nightly and Play Store versions, and I can use External storage on both versions.

**Reply by kevinmcmurtrie** [reply]
I just tried the Play Store version.  Not only is it crippled by SAF, but it uses the wrong SAF when attempting to copy files.  I selected microSD SAF as the destination and imported. It copied to internal SAF then died from running out of space.

I'd say the Play Store version is still a big **no** when it comes to microSD use.

**Reply by FransM** [reply]
if you have a problem with SAF too slow with the play store version it is probably because your phone has insufficient processing power or your SD card is too slow.

**Reply by kevinmcmurtrie** [reply]
This is for a Sony Xperia 1 III (Snapdragon 888) and an LG Velvet 5G (Dimensity 1000C) with a SanDisk Extreme 1TB card (190MB/sec read, 130 MB/sec write).  Do you know of a system that will run 10000 times faster to offset the performance losses of SAF?

I'll keep using the F-Droid version.  I was mostly asking because it's not integrated into any platform for OsmAnd donations.
